src/celestia/CelestiaCoreApplication.cpp

```diff
@@ -0,0 +1,78 @@
+
+#include "CelestiaCoreApplication.h" 
+
+using namespace Eigen;
+
+CelestiaCoreApplication::CelestiaCoreApplication() :
+    showStarsFlag(this),
+    showPlanetsFlag(this),
+    showGalaxiesFlag(this),
+    showDiagramsFlag(this),
+    showCloudMapsFlag(this),
+    showOrbitsFlag(this),
+    showCelestialSphereFlag(this),
+    showNightMapsFlag(this),
+    showAtmospheresFlag(this),
+    showSmoothLinesFlag(this),
+    showEclipseShadowsFlag(this),
+    showStarsAsPointsFlag(this),
+    showRingShadowsFlag(this),
+    showBoundariesFlag(this),
+    showAutoMagFlag(this),
+    showCometTailsFlag(this, "Comet tails enabled", "Comet tails disabled"),
+    showMarkersFlag(this, "Markers enabled", "Markers disabled"),
+    showPartialTrajectoriesFlag(this),
+    showNebulaeFlag(this),
+    showOpenClustersFlag(this),
+    showGlobularsFlag(this),
+    showCloudShadowsFlag(this),
+    showGalacticGridFlag(this),
+    showEclipticGridFlag(this),
+    showHorizonGridFlag(this),
+    showEclipticFlag(this),
+    showTintedIlluminationFlag(this)
+{
+    setAutoMessagesOn();
+}
+
+void CelestiaCoreApplication::gotoSelection(double gotoTime, const Eigen::Vector3f& up, ObserverFrame::CoordinateSystem upFrame) {
+    addToHistory();
+    if (getSimulation()->getFrame()->getCoordinateSystem() == ObserverFrame::Universal)
+        getSimulation()->follow();
+    getSimulation()->gotoSelection(gotoTime, up, upFrame);
+}
+
+void CelestiaCoreApplication::cycleView()
+{
+    do
+    {
+        activeView++;
+        if (activeView == views.end())
+            activeView = views.begin();
+    } while ((*activeView)->type != View::ViewWindow);
+    getSimulation()->setActiveObserver((*activeView)->observer);
+    if (!showActiveViewFrame)  flashFrameStart = currentTime;
+}
+
+bool CelestiaCoreApplication::setLightTravelDelay(bool on) {
+    if (sim->getSelection().body() &&
+        (sim->getTargetSpeed() < 0.99 * astro::speedOfLight))
+    {
+        Vector3d v = sim->getSelection().getPosition(sim->getTime()).offsetFromKm(sim->getObserver().getPosition());
+        lightTravelFlag = !lightTravelFlag;
+        if (on)
```

Comment:
I'd rather rewrite this using ternary operator:
`setLightTravelDelay(on ? v.norm() : -v.norm());